### PR TITLE
Node SDK developer docs

### PR DESCRIPTION
**Pull request by mikeng13**
@haleybash-optimizely @aliabbasrizvi @vraja2 @delikat @mauerbac @jprovine

### CODE REVIEW

File: src/content/server/reference/datafile/0-intro.md [hunk 1]
```diff
@@ -135,7 +135,50 @@ code_examples:
...
+                endOfRange: 4999
+              }
+              {
+                id: "1849205928",
+                key: "variation_b",
+                endOfRange: 9999
+              }
+            ],
+            id: "5928401928",
+            key: "my_experiment"
```
Comment (aliabbasrizvi) [inline src/content/server/reference/datafile/0-intro.md]: The example here is missing the `status`, `audienceIds` and `percentageIncluded` field. Other examples are missing those fields as well. May be other fields also.

**Comment by aliabbasrizvi** [comment]
LGTM other than the 1 comment

File: src/content/server/reference/activation/0-intro.md [hunk 2]
```diff
@@ -49,7 +49,23 @@ code_examples:
...
+  javascript:
+    lang: javascript
+    request: |
+      var experimentKey = 'my_experiment';
+      var userId = 'user123';
+
+      // Conditionally activate an experiment for the provided user
+      optimizely.activate(experimentKey, userId)
+        .then(function(variation) {
+          if (variation == 'variation_a') {
```
Comment (haleybash-optimizely) [inline src/content/server/reference/activation/0-intro.md]: nit: `===`

File: src/content/server/reference/activation/0-intro.md [hunk 3]
```diff
@@ -49,7 +49,23 @@ code_examples:
...
+            // execute code for variation A
+          } else if (variation == 'variation_b') {
```
Comment (haleybash-optimizely) [inline src/content/server/reference/activation/0-intro.md]: nit: `===`

File: src/content/server/reference/activation/1-targeting.md [hunk 4]
```diff
@@ -57,6 +57,25 @@ code_examples:
...
+    lang: javascript
+    request: |
+      var experimentKey = 'my_experiment';
+      var userId = 'user123';
+      var attributes = { 'device': 'iphone', 'ad_source': 'my_campaign' };
+
+      // Conditionally activate an experiment for the provided user
+      optimizely.activate(experimentKey, userId, attributes)
+        .then(function(variation) {
+          if (variation == 'variation_a') {
```
Comment (haleybash-optimizely) [inline src/content/server/reference/activation/1-targeting.md]: nit: `===`

File: src/content/server/reference/activation/1-targeting.md [hunk 5]
```diff
@@ -57,6 +57,25 @@ code_examples:
...
+            // execute code for variation A
+          } else if (variation == 'variation_b') {
```
Comment (haleybash-optimizely) [inline src/content/server/reference/activation/1-targeting.md]: nit: `===`

File: src/content/server/reference/tracking/1-attributes.md [hunk 6]
```diff
@@ -34,6 +34,18 @@ code_examples:
...
+    lang: javascript
+    request: |
+        var eventKey = 'booking_complete';
+        var userId = 'user123';
+        var attributes = { 'device': 'iPhone', 'ad_source': 'my_campaign' };
+
+        // Track a conversion event for the provided user
+        optimizely.track(eventKey, userId, attributes)
+          .then(function() {
+            // conversion tracked
```
Comment (haleybash-optimizely) [inline src/content/server/reference/tracking/1-attributes.md]: ultranit: Could be consistent with `0-intro.md` and say `conversion has been tracked`

**Comment by haleybash-optimizely** [comment]
Looks great! Just a couple nits

**Pull request merged**